File: erlang/numbers/test/numbers_recursive_test.erl
-module(numbers_recursive_test).
-include_lib("eunit/include/eunit.hrl").

recursive_fibonacci_test() ->
    ?assertEqual(55, numbers_recursive:fibonacci(10)).

recursive_factorial_test() ->
    ?assertEqual(120, numbers_recursive:factorial(5)).

recursive_sum_numbers_test() ->
    ?assertEqual(15, numbers_recursive:sum_numbers(5)).

File: fortran/numbers/test/numbers_recursive_test.f90
module numbers_recursive_test
  use funit
  use numbers_recursive
  implicit none

contains
  @test
  subroutine test_fibonacci()
    @assertEqual(fibonacci(5), 5)
  end subroutine test_fibonacci

  @test
  subroutine test_factorial()
    @assertEqual(factorial(5), 120)
  end subroutine test_fibonacci

  @test
  subroutine test_sum_numbers()
    @assertEqual(sum_numbers(5), 15)
  end subroutine test_fibonacci

end module numbers_recursive_test